packages/blockchain-wallet-v4/src/redux/payment/nfts/utils.ts

```diff
@@ -1827,14 +1827,15 @@ async function fungibleTokenApprovals({
     return null
   }
   // eslint-disable-next-line no-console
-  console.log('Not enough ERC20 allowance approved for this trade')
+  console.log('Not enough ERC20 allowance approved for this trade. Approving now...')
 
   // Note: approving maximum amount so this doesnt need to be done again for future trades.
   const txHash = await fungibleTokenInterface.approve(
     proxyAddress,
     ethers.constants.MaxInt256.toString(),
     txnData
   )
+  await txHash.wait()
```

Comment:
Will this always resolve?

Reply:
AFAIK it waits for 1 confirmation. But I think its necessary to prevent that race condition I was talking about (where some other call doesn't think youve approved tokens yet)